lib/Sema/TypeCheckPattern.cpp

```diff
@@ -736,6 +736,74 @@ static bool validateParameterType(ParamDecl *decl, DeclContext *DC,
     }
     decl->getTypeLoc().setType(Ty);
   }
+  // If the param is not a 'let' and it is not an 'inout'.
+  // It must be a 'var'. Provide helpful diagnostics like a shadow copy
+  // in the function body to fix the 'var' attribute.
+  if (!decl->isLet() && !Ty->is<InOutType>()) {
+    auto innermostMethodCtx = dyn_cast_or_null<AbstractFunctionDecl>(DC);
+    if (!innermostMethodCtx || !innermostMethodCtx->hasBody()) {
+      // If there is not function body, just suggest removal of 'var'
+      TC.diagnose(decl->getLetVarInOutLoc(), diag::var_parameter_not_allowed)
+      .fixItRemove(decl->getLetVarInOutLoc());
```

Comment:
If I'm not mistaken, fixItRemove already handles this. In fact the comments there talk about this issue to (quoted below):

>   // If we're removing something (e.g. a keyword), do a bit of extra work to
  // make sure that we leave the code in a good place, without extraneous white
  // space around its hole.  Specifically, check to see there is whitespace
  // before and after the end of range.  If so, nuke the space afterward to keep
  // things consistent.
And the test case `FixIt/fixits-apply.swift` check to make sure that after the fix it there is no surrounding space with the var. So, unless I'm missing something this should be fine?